cmd/podman/main.go

```diff
@@ -29,13 +30,28 @@ import (
 	"github.com/spf13/cobra"
 )
 
+const PodmanSh = "podmansh"
+
 func main() {
 	if reexec.Init() {
 		// We were invoked with a different argv[0] indicating that we
 		// had a specific job to do as a subprocess, and it's done.
 		return
 	}
 
+	if filepath.Base(os.Args[0]) == PodmanSh ||
+		filepath.Base(os.Args[0][1:]) == PodmanSh {
```

Comment:
this can cause a segfault because you try to access first element without checking the len first
```suggestion
		(len(os.Args[0]) > 0 && filepath.Base(os.Args[0][1:]) == PodmanSh) {
```

Reply:
added. PTAL